lib/manageiq/appliance_console/message_configuration_server.rb

```diff
@@ -0,0 +1,225 @@
+        return false unless host_reachable?(@server_hostname, "Message Server")
+
+        true
+      end
+
+      def activate
+        begin
+          create_tools_log_config      # Create the tools log configuration file
+          create_jaas_config           # Create the message server jaas config file
+          create_client_properties     # Create the client.properties config
+          create_logs_directory        # Create the logs directory:
+          configure_firewall           # Open the firewall for message port 9092
+          configure_keystore           # Populate the Java Keystore
+          create_server_properties     # Update the /opt/message/config/server.properties
+          configure_messaging_yaml     # Set up the local message client in case EVM is actually running on this, Message Server
+        rescue AwesomeSpawn::CommandResultError => e
+          say(e.result.output)
+          say(e.result.error)
+          say("")
+          say("Failed to Configure the Message Server- #{e}")
+          return false
+        rescue => e
+          say("Failed to Configure the Message Server- #{e}")
+          return false
+        end
+        true
+      end
+
+      def post_activation
+        say("Starting zookeeper and configure it to start on reboots ...")
+        LinuxAdmin::Service.new("zookeeper").start.enable
+
+        say("Starting kafka and configure it to start on reboots ...")
+        LinuxAdmin::Service.new("kafka").start.enable
+      end
+
+      def ask_for_parameters
+        say("\nMessage Server Parameters:\n\n")
+
+        @username  = ask_for_string("Message Key Username", @username)
+        @password  = ask_for_password("Message Key Password")
+      end
+
+      def show_parameters
+        say("\nMessage Server Configuration:\n")
+        say("Message Server Details:\n")
+        say("  Message Server Hostname:   #{@server_hostname}\n")
+        say("  Message Key Username:      #{@username}\n")
+      end
+
+      private
+
+      def my_hostname
+        LinuxAdmin::Hosts.new.hostname
+      end
+
+      def create_jaas_config
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        content = <<~JAAS
+          KafkaServer {
+            org.apache.kafka.common.security.plain.PlainLoginModule required
+            username=#{@username}
+            password=#{@password}
+            user_admin=#{@password} ;
+          };
+        JAAS
+
+        File.write(@jaas_config_path, content) unless file_found?(@jaas_config_path)
+      end
+
+      def create_logs_directory
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        return true if file_found?(LOGS_DIR)
+
+        FileUtils.mkdir_p(LOGS_DIR)
+        FileUtils.chmod(0o755, LOGS_DIR)
+        FileUtils.chown("kafka", "kafka", LOGS_DIR)
+      end
+
+      def configure_firewall
+        say(__method__.to_s.tr("_", " ").titleize)
+
+        AwesomeSpawn.run!("firewall-cmd --add-port=9092/tcp --permanent") # insecure
```

Comment:
Since we're going through the trouble of setting up SSL, I don't think we should listen on 9092 for insecure connections.

Reply:
@bdunne OK. I had meant to ask you opinion on this. 

Addressed by commit: https://github.com/ManageIQ/manageiq-appliance_console/pull/130/commits/834642c3ea9207676d43da07cf4eff73f94524fa